### ISSUE TITLE
a universal feature importance analysis

### ISSUE THREAD

**Issue by YangR14ustc** [yzhao062/pyod#470]
I wanted to conduct feature importance analysis, but found that many models did not provide feature importance analysis methods except iforest and xgbod .

**Comment by yzhao062** [comment]
that is right...just like most ML algorithms, detection algorithms usually do not have such a feature importance thing except for tree-based methods.

**Comment by YangR14ustc** [comment]
Dear yzhao,
&nbsp; &nbsp; &nbsp; The recent project involves the anomaly detection of multi-dimensional time series and the root cause analysis (anomaly Localization) that causes the anomaly. We have found many methods for anomaly detection of time series, but there are few methods for root cause analysis after judging the anomaly. At present, we use unsupervised algorithms to detect the anomaly, such as the method in pyod,but we do not know what method&nbsp;can be used for root cause analysis. Could you provide the root cause analysis method?
&nbsp; &nbsp; &nbsp;&nbsp;I look forward to hearing from you,thanks.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "yzhao062/pyod"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年1月8日(星期天) 晚上11:04
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [yzhao062/pyod] a universal feature importance analysis (Issue #470)





 
that is right...just like most ML algorithms, detection algorithms usually do not have such a feature importance thing except for tree-based methods.
